PrivateHospitals/backend/PrivateHospitals/PrivateHospital.Migration/Services/AppointmentService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using PrivateHospital.Migration.Dto;
+using PrivateHospital.Migration.Dto.Interfaces;
+using PrivateHospital.Migration.Dto.Repositories;
+using PrivateHospitals.Core.Enum;
+using PrivateHospitals.Core.Models;
+using PrivateHospitals.Core.Models.Users;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PrivateHospital.Migration.Services
+{
+    public class AppointmentService(
+            IRepository<Appointment> _appointmentRepository,
+            IRepository<Doctor> _doctorRepository,
+            IRepository<Patient> _patientRepository
+        )
+    {
+        public async Task SaveAppointmentAsync(AppointmentDto appointmentDto)
+        {
+            if(appointmentDto != null)
+            {
+                var doctor = await _doctorRepository.GetByExternalId(appointmentDto.DoctorExternalId);
+                var patient = await _patientRepository.GetByExternalId(appointmentDto.PatientExternalId);
```

Comment:
Do it differently. It's very heavy operation to SELECT TWICE per appointment.
U can create a full model: appointment with both doctor and patient (all 3 entities fully filled) and then save changes. This way u avoid extra selects on DB and easily work with doctor/patient ids (because ur appointment will just reference doctor and patient)